lib/src/subtree.c

```diff
@@ -997,7 +997,7 @@ void ts_subtree__print_dot_graph(const Subtree *self, uint32_t start_offset,
     ts_subtree_lookahead_bytes(*self)
   );
 
-  if (ts_subtree_is_error(*self) && ts_subtree_child_count(*self) == 0) {
+  if (ts_subtree_is_error(*self) && ts_subtree_child_count(*self) == 0 && self->ptr->lookahead_char != 0) {
     fprintf(f, "\ncharacter: '%c'", self->ptr->lookahead_char);
   }
```

Comment:
Shouldn't we a print a replacement character here e.g.NUL or \0

Reply:
Ueah maybe, well \0 itself wouldn't work but escaping it like "\\0" would if that's what you mean. I initially just thought about the EOF case but there's also some languages where the nul byte is valid in certain spots as well.